source/dotnet/Library/AdaptiveCards/AdaptiveRichTextBlock.cs

```diff
@@ -0,0 +1,124 @@
+#if !NETSTANDARD1_3
+        [XmlAttribute]
+#endif
+        [DefaultValue(typeof(AdaptiveHorizontalAlignment), "left")]
+        public AdaptiveHorizontalAlignment HorizontalAlignment { get; set; }
+
+        /// <summary>
+        ///     Is it allowed for the text to wrap
+        /// </summary>
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+#if !NETSTANDARD1_3
+        [XmlAttribute]
+#endif
+        [DefaultValue(false)]
+        public bool Wrap { get; set; }
+
+        /// <summary>
+        ///     When Wrap is true, you can specify the maximum number of lines to allow the textBlock to use.
+        /// </summary>
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+#if !NETSTANDARD1_3
+        [XmlAttribute]
+#endif
+        [DefaultValue(0)]
+        public int MaxLines { get; set; }
+
+        [JsonConverter(typeof(IgnoreEmptyItemsConverter<AdaptiveParagraph>))]
+#if !NETSTANDARD1_3
+        [XmlElement(typeof(AdaptiveParagraph))]
+#endif
+        public List<AdaptiveParagraph> Paragraphs { get; set; } = new List<AdaptiveParagraph>();
+
+        [JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
+        public class AdaptiveParagraph
+        {
+            [JsonConverter(typeof(IgnoreEmptyItemsConverter<AdaptiveInlineRun>))]
+#if !NETSTANDARD1_3
+            [XmlElement(typeof(AdaptiveTextRun))]
+#endif
+            public List<AdaptiveInlineRun> Inlines { get; set; } = new List<AdaptiveInlineRun>();
+
+            public abstract class AdaptiveInlineRun : AdaptiveTypedElement { }
+
+#if !NETSTANDARD1_3
+            [XmlType(TypeName = AdaptiveTextBlock.TypeName)]
+#endif
+            public class AdaptiveTextRun : AdaptiveInlineRun, ITextElement
+            {
+                public const string TypeName = "TextRun";
+
+                public override string Type { get; set; } = TypeName;
+
+                public AdaptiveTextRun()
+                {
+
+                }
+
+                public AdaptiveTextRun(string text)
+                {
+                    Text = text;
+                }
+
+                /// <summary>
+                ///     The size of the text
+                /// </summary>
+                public AdaptiveTextSize Size { get; set; }
+
+                /// <summary>
+                ///     The weight of the text
+                /// </summary>
+                public AdaptiveTextWeight Weight { get; set; }
+
+                /// <summary>
+                ///     The color of the text
+                /// </summary>
+                public AdaptiveTextColor Color { get; set; }
+
+                /// <summary>
+                ///     Make the text less prominent
+                /// </summary>
+                public bool IsSubtle { get; set; }
+
+                /// <summary>
+                ///     The text to display
+                /// </summary>
+                public string Text { get; set; } = " ";
```

Comment:
>  [](start = 52, length = 1)

is this your test failure?

Reply:
Nope. :( Besides, this is the same default value as for regular text block so I'll keep it around anyways.

---
In reply to: [263149504](https://github.com/Microsoft/AdaptiveCards/pull/2519#discussion_r263149504) [](ancestors = 263149504)